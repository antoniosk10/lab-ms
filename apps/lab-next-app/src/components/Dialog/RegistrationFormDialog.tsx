import TextField from '@/../../packages/lab-fields/src/components/TextField'
import { Box, Button } from '@mui/material'
import { FormProvider, useForm } from 'react-hook-form'

import { SimpleDialog } from './SimpleDialog'

type Props = {
  isOpen: boolean
  onClose: () => void
  onSubmit: (data: Record<string, string>) => void
}

const defaultValues = {
  email: '',
  password: '',
}

export function RegistrationFormDialog({ isOpen, onClose, onSubmit }: Props) {
  const form = useForm({
    defaultValues,
  })

  const { handleSubmit, reset } = form

  const handleClose = () => {
    reset(defaultValues)
    onClose()
  }

  return (
    <FormProvider {...form}>
      <SimpleDialog isOpen={isOpen} title="Sign Up" onClose={handleClose}>
        <Box
          component="form"
          onSubmit={handleSubmit(onSubmit)}
          sx={{ mt: 1, p: 2 }}
        >
          <TextField required={true} id="email" label="Email" name="email" />
          <TextField
            required={true}
            id="password"
            label="Password"
            name="password"
          />
          <TextField
            required={true}
            id="confirmPassword"
            label="Confirm password"
            name="confirmPassword"
          />
          <Button type="submit" variant="contained" sx={{ mt: 3, mb: 2 }}>
            Sign up
          </Button>
        </Box>
      </SimpleDialog>
    </FormProvider>
  )
}
