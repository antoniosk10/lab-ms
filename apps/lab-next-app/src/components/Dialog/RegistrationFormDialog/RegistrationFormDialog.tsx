import { yupResolver } from '@hookform/resolvers/yup'
import { Box, Button, Typography } from '@mui/material'
import { red as redColor } from '@mui/material/colors'
import TextField from '@packages/lab-fields/src/components/TextField'
import { useState } from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import * as yup from 'yup'

import { Notification } from '../../Notification'
import { SimpleDialog } from '../SimpleDialog/SimpleDialog'

import { AuthUserData } from '@/src/types'

type Props = {
  isOpen: boolean
  onClose: () => void
  onSubmit: (data: AuthUserData) => void
}

type FormData = {
  email: string
  password: string
  confirmPassword: string
}

const defaultValues = {
  email: '',
  password: '',
  confirmPassword: '',
}

const schema = yup
  .object({
    email: yup.string().required('Email is required'),
    password: yup.string().required('Password is required'),
    confirmPassword: yup
      .string()
      .oneOf([yup.ref('password')], 'Passwords must match'),
  })
  .required()

export function RegistrationFormDialog({ isOpen, onClose, onSubmit }: Props) {
  const [isNotificationOpen, setNotificationOpen] = useState(false)

  const form = useForm({
    defaultValues,
    resolver: yupResolver(schema),
    reValidateMode: 'onSubmit',
  })

  const {
    handleSubmit,
    reset,
    formState: { errors },
  } = form

  const handleClose = () => {
    reset(defaultValues)
    onClose()
  }

  const handleOnSubmit = ({ email, password }: FormData) => {
    onSubmit({ email, password })
    setNotificationOpen(true)
  }

  const handleNotificationClose = () => {
    setNotificationOpen(false)
  }

  return (
    <>
      <FormProvider {...form}>
        <SimpleDialog isOpen={isOpen} title="Sign Up" onClose={handleClose}>
          <Box
            component="form"
            onSubmit={handleSubmit(handleOnSubmit)}
            sx={{ mt: 1, p: 2 }}
          >
            <TextField required={true} id="email" label="Email" name="email" />
            <TextField
              required={true}
              id="password"
              label="Password"
              name="password"
              type="password"
            />
            <TextField
              required={true}
              id="confirmPassword"
              label="Confirm password"
              name="confirmPassword"
              type="password"
            />

            {errors.confirmPassword?.message && (
              <Typography color={redColor[600]}>
                {errors.confirmPassword?.message}
              </Typography>
            )}

            <Button type="submit" variant="contained" sx={{ mt: 3, mb: 2 }}>
              Sign up
            </Button>
          </Box>
        </SimpleDialog>
      </FormProvider>
      <Notification
        isOpen={isNotificationOpen}
        text="You've been registered. Please Sign In"
        onClose={handleNotificationClose}
      />
    </>
  )
}
