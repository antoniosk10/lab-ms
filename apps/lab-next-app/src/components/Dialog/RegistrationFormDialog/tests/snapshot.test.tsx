import { render } from '@testing-library/react'

import { RegistrationFormDialog } from '../RegistrationFormDialog'

it('renders RegistrationFormDialog unchanged', () => {
  const { baseElement } = render(
    <RegistrationFormDialog
      isOpen={true}
      onClose={jest.fn}
      onSubmit={jest.fn}
    />
  )
  expect(baseElement).toMatchSnapshot()
})
