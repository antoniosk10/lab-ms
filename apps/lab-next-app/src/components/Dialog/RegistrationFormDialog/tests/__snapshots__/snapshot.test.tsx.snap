// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RegistrationFormDialog unchanged 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiDialogTitle-root css-xrt2rx-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          Sign Up
        </h2>
        <form
          class="MuiBox-root css-12987b1"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              Email
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Email
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="confirmPassword"
              id="confirmPassword-label"
            >
              Confirm password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="confirmPassword"
                name="confirmPassword"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Confirm password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1n7zltw-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Sign up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
